### DISCUSSION TITLE
What does XamlCheckProcessRequirements do?

### DISCUSSION THREAD

**Discussion by amtopel** [microsoft/microsoft-ui-xaml#8445] [Q&A]
I'm wondering what `XamlCheckProcessRequirements`, which precedes the message loop inside the XAML-generated main function, does.  Like what does it look for, and what does it do on failure?

Also, I've come across an [example](https://stackoverflow.com/a/72705577) of a custom main function that removes that function call entirely.  Is that okay to do?

**Comment by DarranRowe** [comment]
IIRC, it basically does nothing. But what it does is put a crash point into main if, for some reason, the process starts without bootstrapping the Windows App Runtime. It will cause a crash in main which is easy to diagnose rather than some less obvious failure when trying to instantiate a Windows Runtime object.

**Reply by amtopel** [reply]
Okay, cool.  Thanks for the info.
